apps/web/app/(basenames)/name/[username]/ProfileProviders.tsx

```diff
@@ -1,21 +1,40 @@
 'use client';
 
+import { BaseName } from '@coinbase/onchainkit/identity';
 import AnalyticsProvider from 'apps/web/contexts/Analytics';
 import UsernameProfileProvider from 'apps/web/src/components/Basenames/UsernameProfileContext';
+import { formatBaseEthDomain, USERNAME_DOMAINS } from 'apps/web/src/utils/usernames';
+import { Address } from 'viem';
+import { base, baseSepolia } from 'viem/chains';
 
 // Do not change this unless you know what you're doing (it'll break analytics)
 const usernameProfileAnalyticContext = 'username_profile';
 
 export default function ProfileProviders({
   children,
   username,
+  address,
 }: {
   children: React.ReactNode;
-  username: string;
+  username: BaseName;
+  address: Address;
 }) {
+  let formattedUsername = username;
+
+  // Resolve domainless name to .base.eth
+  if (
```

Comment:
Should this use `formatDefaultUsername`?

Reply:
ah good catch, actually this is not even needed since the page should pass the correct domain.